src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractExtensionMethodImportCompletionProvider.cs

```diff
@@ -21,8 +21,8 @@ internal abstract class AbstractExtensionMethodImportCompletionProvider : Abstra
     {
         protected abstract string GenericSuffix { get; }
 
-        protected override bool ShouldProvideCompletion(Document document, SyntaxContext syntaxContext)
-            => syntaxContext.IsRightOfNameSeparator && IsAddingImportsSupported(document);
+        protected override bool ShouldProvideCompletion(CompletionContext completionContext, SyntaxContext syntaxContext, Document document)
+            => syntaxContext.IsRightOfNameSeparator && IsAddingImportsSupported(syntaxContext.Workspace, document, completionContext.Options.GetOption(CompletionServiceOptions.DisallowAddingImports));
```

Comment:
consider passing in `completionContext.Options` and having IsAddingImportsSupported check for the appropriate option within that set insetad of all callers having to do it.

Reply:
We don't have Options in all call sites.